ha/util/actuator_req_manager.py

```diff
@@ -0,0 +1,246 @@
+            return True
+
+        if not self._is_resp_received:
+            Log.error(f"Actuator response not received; enclosure shutdown failed on node {node_name}")
+        else:
+            Log.error(f"Unable to shutdown enclosure on node {node_name}")
+
+        self._is_resp_received = self._encl_shutdown_successful = False
+        raise ClusterManagerError("Failed to shutdown the enclosure")
+
+    def _create_req(self, target_node_name: str) -> str:
+        """
+        Create actualtor request for enclosure stop
+
+        Args:
+            target_node_name : node on which enclosure is to be stopped
+
+        Return:
+            created actuator request
+        """
+
+        with open(ACTUATOR_SCHEMA, 'r') as actuator_req_schema_file:
+            actuator_req = json.load(actuator_req_schema_file)
+
+        # Get details for current node
+        node_name = Conf.get(const.HA_GLOBAL_INDEX, f"CLUSTER_MANAGER{_DELIM}local_node")
+
+        key_val  = self._conf_store.get(f"{PVTFQDN_TO_NODEID_KEY}/{node_name}")
+        _, node_id = key_val.popitem()
+
+        system_health = SystemHealth(self._conf_store)
+        health_manager = SystemHealthManager(self._conf_store)
+        key = system_health._prepare_key(const.COMPONENTS.NODE_MAP.value, node_id=node_id)
+
+        node_map_val = health_manager.get_key(key)
+        if node_map_val is None:
+            raise ClusterManagerError("Failed to fetch node_map value")
+
+        node_map_dict = ast.literal_eval(node_map_val)
+        self._generate_uuid()
+
+        # Populate the actuator request schema
+        actuator_req[ACTUATOR_ATTRIBUTES.TIME] = str(int(time.time()))
+        actuator_req[ACTUATOR_ATTRIBUTES.MESSAGE][ACTUATOR_ATTRIBUTES.REQUEST_PATH][ACTUATOR_ATTRIBUTES.SITE_ID] = node_map_dict[NODE_MAP_ATTRIBUTES.SITE_ID.value]
+        actuator_req[ACTUATOR_ATTRIBUTES.MESSAGE][ACTUATOR_ATTRIBUTES.REQUEST_PATH][ACTUATOR_ATTRIBUTES.RACK_ID] = node_map_dict[NODE_MAP_ATTRIBUTES.RACK_ID.value]
+        actuator_req[ACTUATOR_ATTRIBUTES.MESSAGE][ACTUATOR_ATTRIBUTES.REQUEST_PATH][ACTUATOR_ATTRIBUTES.NODE_ID] = node_id
+        actuator_req[ACTUATOR_ATTRIBUTES.MESSAGE][ACTUATOR_ATTRIBUTES.TARGET_NODE_ID] = target_node_name
+        actuator_req[ACTUATOR_ATTRIBUTES.MESSAGE][ACTUATOR_ATTRIBUTES.HEADER][ACTUATOR_ATTRIBUTES.UUID] = self._uuid
+
+        actuator_req_schema_file.close()
+        req = json.dumps(actuator_req)
+
+        return req
+
+    def _send_req(self, req):
+        """
+        Send the created request to "monitor" on message bus
+
+        Args:
+            req : created actuator request
+        """
+
+        self.req_message_type = Conf.get(const.HA_GLOBAL_INDEX, f"ACTUATOR_MANAGER{_DELIM}req_message_type")
+        self.producer_id = Conf.get(const.HA_GLOBAL_INDEX, f"ACTUATOR_MANAGER{_DELIM}producer_id")
+        self.producer =  MessageBus.get_producer(self.producer_id, self.req_message_type)
+
```

Comment:
handle case for if producer is empty

Reply:
get_producer()  creates a new producer object and returns it. So it can not be empty